subgraphs/aave-forks/src/helpers.ts

```diff
@@ -1059,14 +97,125 @@ export function getBorrowBalance(
 
   // get account's balance of stable debt
   if (market._sToken) {
-    const sTokenContract = AToken.bind(Address.fromString(market._sToken!));
+    const sTokenContract = AToken.bind(Address.fromBytes(market._sToken!));
     const trySDebtTokenBalance = sTokenContract.try_balanceOf(account);
     sDebtTokenBalance = trySDebtTokenBalance.reverted
       ? BIGINT_ZERO
       : trySDebtTokenBalance.value;
   }
 
-  const totalDebt = sDebtTokenBalance.plus(vDebtTokenBalance);
+  // const totalDebt = sDebtTokenBalance.plus(vDebtTokenBalance);
+
+  return [sDebtTokenBalance, vDebtTokenBalance];
+}
+
+export function getCollateralBalance(market: Market, account: Address): BigInt {
+  let collateralBalance = BIGINT_ZERO;
+  const aTokenContract = AToken.bind(Address.fromBytes(market.outputToken!));
+  const balanceResult = aTokenContract.try_balanceOf(account);
+  if (balanceResult.reverted) {
+    log.warning(
+      "[getCollateralBalance]failed to get aToken {} balance for {}",
+      [market.outputToken!.toHexString(), account.toHexString()]
+    );
+    return collateralBalance;
+  }
+
+  let exchangeRate = BIGDECIMAL_ONE;
+  if (!market.exchangeRate) {
+    log.info(
+      "[getCollateralBalance]market {} exchange rate not set, default to 1.0",
+      [market.id.toHexString()]
+    );
+    return collateralBalance;
+  }
+  exchangeRate = market.exchangeRate!;
+
+  collateralBalance = bigDecimalToBigInt(
+    balanceResult.value.toBigDecimal().times(exchangeRate)
+  );
+  return collateralBalance;
+}
+
+export function getOrCreateFlashloanPremium(
+  procotolData: ProtocolData
+): _FlashLoanPremium {
+  let flashloanPremium = _FlashLoanPremium.load(procotolData.protocolID);
+  if (!flashloanPremium) {
+    flashloanPremium = new _FlashLoanPremium(procotolData.protocolID);
+    flashloanPremium.premiumRateTotal = BIGDECIMAL_ZERO;
+    flashloanPremium.premiumRateToProtocol = BIGDECIMAL_ZERO;
+    flashloanPremium.premiumUSDToDeduct = BIGDECIMAL_ZERO;
+    flashloanPremium.save();
+  }
+  return flashloanPremium;
+}
+
+export function getOrCreateAccountDebtBalance(
+  market: Market,
+  account: Address
+): _AccountDebtBalance {
+  const id = account.concat(market.id);
+  let accountDebtBalance = _AccountDebtBalance.load(id);
+  if (!accountDebtBalance) {
+    accountDebtBalance = new _AccountDebtBalance(id);
+    accountDebtBalance.sTokenBalance = BIGINT_ZERO;
+    accountDebtBalance.vTokenBalance = BIGINT_ZERO;
+    accountDebtBalance.save();
+  }
+  return accountDebtBalance;
+}
+
+export function storePrePauseState(market: Market): void {
+  market._prePauseState = [
+    market.isActive,
+    market.canUseAsCollateral,
+    market.canBorrowFrom,
+  ];
+  market.save();
+}
+
+export function restorePrePauseState(market: Market): void {
+  if (!market._prePauseState || market._prePauseState!.length !== 3) {
+    log.warning(
```

Comment:
log an error instead. Something would probably be wrong with the code if this happened

Reply:
fixed